### PR TITLE
Update changelog for 0.0.4 release

### PR DESCRIPTION
**Pull request by natecook1000**

### CODE REVIEW

**Comment by natecook1000** [comment]
@swift-ci Please test

**Comment by natecook1000** [comment]
@swift-ci Please test macOS platform

**Review by timvermeulen — APPROVED** [review]
🎉

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,33 @@ package updates, you can specify your package dependency using
...
+### Fixes
+
+- Methods and computed properties are more consistently marked as inlinable, 
+  resolving a performance regression. 
+- The `Stride` type now efficiently calculates distances between positions,
+  supported by the underlying collection.
+- Better test coverage and improved diagnostics for comparing sequences.
+- Fixed links and improved documentation.
+
+The 0.0.3 release includes contributions from [bjhomer], [danielctull],
```
Comment (LucianoPAlmeida) [inline CHANGELOG.md:42]: Hey @natecook1000 :) 
Should this be 0.0.4?

Comment (natecook1000) [reply CHANGELOG.md]: Gah, thanks!

**Comment by natecook1000** [comment]
@swift-ci Please test

**Pull request merged**